modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/node/DuplicateNodeCommand.java

```diff
@@ -225,20 +225,15 @@ private String resolveNewNodeName( final Node existingNode )
         // Process as file name as it is so for images
         String newNodeName = DuplicateValueResolver.fileName( existingNode.name().toString() );
 
-        boolean resolvedUnique = false;
-
-        while ( !resolvedUnique )
+        while ( true )
```

Comment:
we may likely drop this optimization and instead catch NodeAlreadyExists with retry.